.github/workflows/linters.yml

```
@@ -50,7 +49,7 @@ jobs:
       - name: Lint C/C++ files
         run: make lint-cpp
   format-cpp:
-    if: ${{ github.event.pull_request.draft == false && github.base_ref == 'master' }}
+    if: ${{ github.event.pull_request.draft == false && github.event.pull_request.base.ref == github.event.pull_request.base.repo.default_branch }}
```

What will happen when it's a push instead of a pull request?

---

I've just tested on https://github.com/nodejs/node-auto-test and it looks like the job is no longer skipped on push event with this change: https://github.com/nodejs/node-auto-test/runs/6953844826?check_suite_focus=true

I think we need an actual check (or keep the default branch "hardcoded" in the action) (I've tested it in https://github.com/nodejs/node-auto-test/actions/runs/2523331560)
```suggestion
    if: ${{ github.event.pull_request.draft == false && github.event.pull_request && github.base_ref == github.event.pull_request.base.repo.default_branch }}
```

---

After some research, I found we could simplify this expression to :

```yaml
    if: ${{ github.event.pull_request.draft == false && github.base_ref == github.repository.default_branch  }}
```

see [here](https://github.com/nodejs/node/blob/5fad0b93667ffc6e4def52996b9529ac99b26319/.github/workflows/notify-force-push.yml#L21) for an example on node repo.

---

I don't think that's correct, I've tested that on https://github.com/nodejs/node-auto-test/actions/runs/2523329579 and the job wasn't skipped.

---

OMG, GitHub action is complicated. what if we just take the "keep the default branch hardcoded in the action" approach? I don't think the default branch will change in the next 5 years. I think the hardcode way is simple.

---

I've updated my suggestion to simplify it a bit, I'm OK with keeping the default branch name hardcoded although it's definitely not as nice.

---

> I've updated my suggestion to simplify it a bit and make it more readable, I'm OK with keeping the default branch name hardcoded although it's definitely not as nice.

Great. Now the code is simpler. Suggestion committed. Please take another test. Thanks you.

---

A question: Why do we need to add `github.event.pull_request &&` check? It is not here before this pr and before this pr this workflow works normally.

---

Not sure, but at least now it's way clearer that we intend to skip when not on a PR.

---

Yep. It will be better now given we do the explicit check if `github.event.pull_request` is empty.